cogs/logging.py

```diff
@@ -41,15 +43,105 @@ async def on_member_ban(self,guild,user):
+                    allowed_mentions=AllowedMentions(
+                        users=[Object(id=before.id)]
+                    )
+                )
+            elif after.timed_out_until is None:
+                await channel.send(
+                    content=
+                        f"# Member timeout removed\n"
+                        f"**Member:** `{before.id}` <@{before.id}>\n"
+                        f"**Was timed out until:** <t:{int(before.timed_out_until.timestamp())}:F>",
+                    allowed_mentions=AllowedMentions(
+                        users=[Object(id=before.id)]
+                    )
+                )
+            else: # Not sure if this can happen
+                await channel.send(
+                    content=
+                        f"# Member timeout changed\n"
+                        f"**Member:** `{before.id}` <@{before.id}>\n"
+                        f"**Was timed out until:** <t:{int(before.timed_out_until.timestamp())}:F>"
+                        f"**Now timed out until:** <t:{int(after.timed_out_until.timestamp())}:F>",
+                    allowed_mentions=AllowedMentions(
+                        users=[Object(id=before.id)]
+                    )
+                )
+
 
-    # TODO implement delete messages
     @commands.Cog.listener()
     async def on_message_delete(self,message):
-        pass
+        channel = self.logging_channels[message.guild.id]
+
+        # Do not log deletions in the logging channel itself
+        if message.channel.id == channel.id:
+            return
+        
+        content_notice = ""
+
+        # Replacing with U+2035 REVERSED PRIME
+        sanitized_content = message.content.replace("```", "‵‵‵") 
+        character_limit = 1500
+        if len(sanitized_content) > character_limit:
+            sanitized_content = sanitized_content[:character_limit]
+            content_notice += f" (trimmed at {character_limit})"
+
+        types = []
+        if message.flags.crossposted:
+            types.append("crossposted")
+        if message.flags.is_crossposted:
+            types.append("is_crossposted")
+        if message.flags.source_message_deleted:
+            types.append("source_message_deleted")
+        if message.flags.suppress_embeds:
+            types.append("suppress_embeds")
+        if message.flags.urgent: # shouldn't happen?
+            types.append("urgent")
+        if message.flags.has_thread:
+            types.append("has_thread")
+        if message.flags.ephemeral: # shouldn't happen?
+            types.append("ephemeral")
+        if message.flags.loading: # shouldn't happen?
+            types.append("loading")
+        if message.flags.failed_to_mention_some_roles_in_thread:
+            types.append("failed_to_mention_some_roles_in_thread")
+        if message.flags.silent:
+            types.append("silent")
+        # Needs discord.py >= 2.3
+        # if message.flags.voice:
+        #     types.append("voice")
```

Comment:
I feel like there might be a cleaner way to do this? If not, then I'd recommend putting it in a separate function to improve the code flow.